File: TCP  IP Server Client Example Part 7 (Chat Application)/[VB]-TCP  IP Server Client Example Part 7 (Chat Application)/VB/ServerClientChat1.3/Forms/FrmAzureDatabaseLogin.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.IO
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace ServerClientChat1._3.Forms
	Partial Public Class FrmAzureDatabaseLogin
		Inherits Form

		Public Property ServerName() As String
		Public Property UserName() As String
		Public Property Password() As String

		Public Sub New()
			InitializeComponent()
			ProcessCheckedState()

#If DEBUG Then ' Only show the buttons we need depending upon the running mode.
			btnLoad.Enabled = True
			btnLoad.Visible = True
			btnCancel.Enabled = False
			btnCancel.Visible = False
			btnConnect.Visible = False
			btnConnect.Enabled = False

#Else
			btnLoad.Enabled = True
			btnLoad.Visible = True
			btnCancel.Enabled = True
			btnCancel.Visible = True
			btnConnect.Visible = True
			btnConnect.Enabled = True
#End If
		End Sub

		''' <summary>
		''' Validate the Server name
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub tbServername_Validating(ByVal sender As Object, ByVal e As CancelEventArgs) Handles tbServername.Validating
			ServerName = tbServername.Text.Replace("tcp:", "").Trim() ' Remove the tcp: if there and any extra spaces
#If DEBUG Then
			' Get Results from the Load Button
			' If we do not use the Compile Time constants the ErrorProvider stops us from using the Load Button.
#Else
			 If String.IsNullOrWhiteSpace(ServerName) Then
				errorProvider1.SetError(tbServername, "Server name cannot be empty")
				e.Cancel = True ' Don't move to next control
			Else
				If Not ServerName.Contains(","c) Then
					errorProvider1.SetError(tbServername, "Add , then the port to be used.") ' We should check that the name does not end in , but I will leave that to you to work out how to validate that.
					e.Cancel = True ' Don't move to next control
				Else
					errorProvider1.Clear() ' We don't need to show the Error icon now if it is showing
					e.Cancel = False ' move to next control
				End If
			End If
#End If
		End Sub

		''' <summary>
		''' Validate the User name
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub tbUsername_Validating(ByVal sender As Object, ByVal e As CancelEventArgs) Handles tbUsername.Validating
#If DEBUG Then
			' Get values from Load Button
#Else
			 If String.IsNullOrWhiteSpace(tbUsername.Text) Then
				errorProvider1.SetError(tbUsername, "User name cannot be empty")
				e.Cancel = True ' Don't move to next control
			Else
				UserName = tbUsername.Text.Trim() ' Make sure there are no extra spaces.
				errorProvider1.Clear() ' We don't need to show the Error icon now if it is showing
				e.Cancel = False ' move to next control
			End If
#End If
		End Sub

		''' <summary>
		''' Validate the User's Password
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub tbPassword_Validating(ByVal sender As Object, ByVal e As CancelEventArgs) Handles tbPassword.Validating
#If DEBUG Then
			' Get Values from Load button.
#Else
			If String.IsNullOrWhiteSpace(tbPassword.Text) Then
				errorProvider1.SetError(tbPassword, "Password name cannot be empty")
				e.Cancel = True ' Don't move to next textbox
			Else
				Password = tbPassword.Text.Trim() ' Make sure there are no extra spaces.
				errorProvider1.Clear() ' We don't need to show the Error icon now if it is showing
				e.Cancel = False ' move to next control
			End If
#End If
		End Sub

		''' <summary>
		''' User canceled connection
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub btnCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCancel.Click
			Me.Close()
		End Sub

		''' <summary>
		''' Make the Connection
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub btnConnect_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnConnect.Click
			DialogResult = System.Windows.Forms.DialogResult.OK
			Me.Close()
		End Sub

		Private Sub cbRememberForThisSession_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbRememberForThisSession.CheckedChanged
			ProcessCheckedState()
		End Sub

		Private Sub ProcessCheckedState()
			If cbRememberForThisSession.Checked Then
				tbServername.Text = ServerName
				tbUsername.Text = UserName
				tbPassword.Text = Password
			End If
		End Sub

		''' <summary>
		''' Loads the User Connection Details from a File called ChatAzureDB.txt located in My Documents
		''' The file contains three lines
		''' Line 1 Server Details: xxxxxxxxxx.database.windows.net,1433
		''' Line 2 User Name     : YourUserName@xxxxxxxxxx
		''' Line 3 Plain text Password: xxxxxxxxxxxxxxxxxx
		''' 
		''' This is only to aid us in Debugging and SHOULD NOT BE USED IN PRODUCTION
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub btnLoad_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnLoad.Click
			Try
				Dim path As String = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments)
				path = System.IO.Path.Combine(path, "ChatAzureDB.txt")

				Dim fs As New FileStream(path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite)
				Dim sr As New StreamReader(fs)

				ServerName = sr.ReadLine()
				UserName = sr.ReadLine()
				Password = sr.ReadLine()

				sr.Close()
				fs.Close()
				DialogResult = System.Windows.Forms.DialogResult.OK
				Me.Close()
			Catch ex As Exception
				MessageBox.Show("Load Configuration File: " & ex.Message)
			End Try

		End Sub
	End Class
End Namespace

File: TCP  IP Server Client Example Part 7 (Chat Application)/[C#][VB]-TCP  IP Server Client Example Part 7 (Chat Application)/C# and VB/ServerClientChat1.3/Forms/FrmAzureDatabaseLogin.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ServerClientChat1._3.Forms
{
    public partial class FrmAzureDatabaseLogin : Form
    {

        public string ServerName { get; set; }
        public string UserName { get; set; }
        public string Password { get; set; }

        public FrmAzureDatabaseLogin()
        {
            InitializeComponent();
            ProcessCheckedState();

#if DEBUG   // Only show the buttons we need depending upon the running mode.
            btnLoad.Enabled = true;
            btnLoad.Visible = true;
            btnCancel.Enabled = false;
            btnCancel.Visible = false;
            btnConnect.Visible = false;
            btnConnect.Enabled = false;

#else
            btnLoad.Enabled = true;
            btnLoad.Visible = true;
            btnCancel.Enabled = true;
            btnCancel.Visible = true;
            btnConnect.Visible = true;
            btnConnect.Enabled = true;
#endif
        }

        /// <summary>
        /// Validate the Server name
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void tbServername_Validating(object sender, CancelEventArgs e)
        {
            ServerName = tbServername.Text.Replace("tcp:", "").Trim(); // Remove the tcp: if there and any extra spaces
#if DEBUG
            // Get Results from the Load Button
            // If we do not use the Compile Time constants the ErrorProvider stops us from using the Load Button.
#else
             if (string.IsNullOrWhiteSpace(ServerName))
            {
                errorProvider1.SetError(tbServername, "Server name cannot be empty");
                e.Cancel = true; // Don't move to next control
            }
            else
            {
                if (!ServerName.Contains(','))
                {
                    errorProvider1.SetError(tbServername, "Add , then the port to be used."); // We should check that the name does not end in , but I will leave that to you to work out how to validate that.
                    e.Cancel = true; // Don't move to next control
                }
                else
                {
                    errorProvider1.Clear(); // We don't need to show the Error icon now if it is showing
                    e.Cancel = false; // move to next control
                }
            }
#endif
        }

        /// <summary>
        /// Validate the User name
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void tbUsername_Validating(object sender, CancelEventArgs e)
        {
#if DEBUG
            // Get values from Load Button
#else
             if (string.IsNullOrWhiteSpace(tbUsername.Text))
            {
                errorProvider1.SetError(tbUsername, "User name cannot be empty");
                e.Cancel = true; // Don't move to next control
            }
            else
            {
                UserName = tbUsername.Text.Trim(); // Make sure there are no extra spaces.
                errorProvider1.Clear(); // We don't need to show the Error icon now if it is showing
                e.Cancel = false; // move to next control
            }
#endif
        }

        /// <summary>
        /// Validate the User's Password
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void tbPassword_Validating(object sender, CancelEventArgs e)
        {
#if DEBUG
            // Get Values from Load button.
#else
            if (string.IsNullOrWhiteSpace(tbPassword.Text))
            {
                errorProvider1.SetError(tbPassword, "Password name cannot be empty");
                e.Cancel = true; // Don't move to next textbox
            }
            else
            {
                Password = tbPassword.Text.Trim(); // Make sure there are no extra spaces.
                errorProvider1.Clear(); // We don't need to show the Error icon now if it is showing
                e.Cancel = false; // move to next control
            }
#endif
        }

        /// <summary>
        /// User canceled connection
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void btnCancel_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        /// <summary>
        /// Make the Connection
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void btnConnect_Click(object sender, EventArgs e)
        {
            DialogResult = System.Windows.Forms.DialogResult.OK;
            this.Close();
        }

        private void cbRememberForThisSession_CheckedChanged(object sender, EventArgs e)
        {
            ProcessCheckedState();
        }

        private void ProcessCheckedState()
        {
            if (cbRememberForThisSession.Checked)
            {
                tbServername.Text = ServerName;
                tbUsername.Text = UserName;
                tbPassword.Text = Password;
            }
        }

        /// <summary>
        /// Loads the User Connection Details from a File called ChatAzureDB.txt located in My Documents
        /// The file contains three lines
        /// Line 1 Server Details: xxxxxxxxxx.database.windows.net,1433
        /// Line 2 User Name     : YourUserName@xxxxxxxxxx
        /// Line 3 Plain text Password: xxxxxxxxxxxxxxxxxx
        /// 
        /// This is only to aid us in Debugging and SHOULD NOT BE USED IN PRODUCTION
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void btnLoad_Click(object sender, EventArgs e)
        {
            try
            {
                string path = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
                path = Path.Combine(path, "ChatAzureDB.txt");

                FileStream fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
                StreamReader sr = new StreamReader(fs);

                ServerName = sr.ReadLine();
                UserName = sr.ReadLine();
                Password = sr.ReadLine();

                sr.Close();
                fs.Close();
                DialogResult = System.Windows.Forms.DialogResult.OK;
                this.Close();
            }
            catch (Exception ex)
            {
                MessageBox.Show("Load Configuration File: " + ex.Message);
            }

        }
    }
}
